src/main/java/net/sf/rails/game/special/SpecialSingleTileLay.java

```diff
@@ -0,0 +1,164 @@
+package net.sf.rails.game.special;
+
+import net.sf.rails.common.LocalText;
+import net.sf.rails.common.parser.ConfigurationException;
+import net.sf.rails.common.parser.Tag;
+import net.sf.rails.game.*;
+import net.sf.rails.util.Util;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class SpecialSingleTileLay extends SpecialTileLay {
+    /**
+     * Used by Configure (via reflection) only
+     *
+     * @param parent Railsitem this belongs
+     * @param id this Item has
+     */
+    public SpecialSingleTileLay(RailsItem parent, String id) {
+        super(parent, id);
+    }
+
+    @Override
+    public void configureFromXML(Tag tag) throws ConfigurationException {
+        super.configureFromXML(tag);
+
+        Tag tileLayTag = tag.getChild("SpecialSingleTileLay");
+        if (tileLayTag == null) {
+            throw new ConfigurationException("<SpecialSingleTileLay> tag missing");
+        }
+
+        locationCodes = tileLayTag.getAttributeAsString("location");
+        if (!Util.hasValue(locationCodes))
+            throw new ConfigurationException("SpecialSingleTileLay: location missing");
+
+        tileId = tileLayTag.getAttributeAsString("tile", null);
+
+        String coloursString = tileLayTag.getAttributeAsString("colour");
+        if (Util.hasValue(coloursString)) {
+            tileColours = coloursString.split(",");
+        }
+
+        name = tileLayTag.getAttributeAsString("name");
+
+        extra = tileLayTag.getAttributeAsBoolean("extra", extra);
+        free = tileLayTag.getAttributeAsBoolean("free", free);
+        connected = tileLayTag.getAttributeAsBoolean("connected", connected);
+        discount = tileLayTag.getAttributeAsInteger("discount", discount);
+
+        if (tileId != null) {
+            description = LocalText.getText("LayNamedTileInfo",
+                    tileId,
+                    name != null ? name : "",
+                            locationCodes,
+                            (extra ? LocalText.getText("extra"):LocalText.getText("notExtra")),
+                            (free ? LocalText.getText("noCost") : discount != 0 ? LocalText.getText("discount", discount) :
+                                LocalText.getText("normalCost")),
+                            (connected ? LocalText.getText("connected") : LocalText.getText("unconnected"))
```

Comment:
Can we split this in multiple statements? This is quite a long statement I think it would be easier to understand if it is split in multiple statements

Reply:
Thats just the original implementation move down in the singleclass, but yes...